### DISCUSSION TITLE
Question regarding the usage of `DicomInputStream.readDatasetUntilPixelData()`

### DISCUSSION THREAD

**Discussion by Zhenye-Na** [dcm4che/dcm4che#1396] [Q&A] [answered]
I am wondering what will be the output of `DicomInputStream.getPosition()` after invoking the above method ? Will it be the file pointer position before reaching to the pixelData tag ?

Also, what will be returned if there is no PixelData Tag in the DICOM input file ? Do you happen to know if the total DICOM file size will be returned in this case?

https://github.com/dcm4che/dcm4che/issues/835#issuecomment-1927887557

**Comment by simriska** [comment, marked as answer]
If there is no PixelData you will receive full dataset. 
The position in case of existing PixelData is BEHIND the header of the Pixeldata tag, so 8 bytes further.

**Reply by Zhenye-Na** [reply]
Hello @simriska 

Thanks for the reply here! Since you explicitly mentioned that `existing PixelData is BEHIND the header of the Pixeldata tag`, I am new to this domain and does it mean that there are chances or cases that `existing PixelData is BEFORE the header of the Pixeldata tag`?

Also I am wondering, is `PixelData` kind of the last tag in the whole DICOM file ?

Thank you!

**Reply by simriska** [reply]
See here https://dicom.nema.org/dicom/2013/output/chtml/part05/chapter_7.html. The parsing will stop before the value field.
PixelData is a tag with very high number. As the tags are sorted, mostly it is at the end of the file, but there are cases of tags behind (mostly private vendor tags).
